src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +91,300 @@ const NotebookLauncher = _.flow(
+        h(Fragment, [
+          h(PreviewHeader, { queryParams, cluster, refreshClusters, notebookName, workspace, readOnlyAccess: !(Utils.canWrite(accessLevel) && canCompute), onCreateCluster: () => setCreateOpen(true) }),
+          h(NotebookPreviewFrame, { notebookName, workspace })
+        ]),
+      mode && h(ClusterKicker, { cluster, refreshClusters, onNullCluster: () => { setCreateOpen(true) } }),
+      mode && h(ClusterStatusMonitor, { cluster, onClusterStoppedRunning: () => { chooseMode(undefined) } }),
+      h(NewClusterModal, {
+        isOpen: createOpen,
+        namespace, currentCluster: cluster,
+        onDismiss: () => {
+          chooseMode(undefined)
+          setCreateOpen(false)
+        },
+        onSuccess: withErrorReporting('Error creating cluster', async promise => {
+          setCreateOpen(false)
+          setBusy(true)
+          await promise
+          await refreshClusters()
+          setBusy(false)
+        })
+      }),
+      busy && spinnerOverlay
     ])
-})
+  })
 
-const ReadOnlyMessage = ({ notebookName, workspace, workspace: { canCompute, workspace: { namespace, name } } }) => {
-  const [copying, setCopying] = useState(false)
+const FileInUseModal = ({ onDismiss, onCopy, onPlayground, namespace, name, bucketName, lockedBy, canShare }) => {
+  const [lockedByEmail, setLockedByEmail] = useState(null)
 
-  return div({ style: { padding: '1rem 2rem', display: 'flex', alignItems: 'center' } }, [
-    div({ style: { fontSize: 16, fontWeight: 'bold', position: 'absolute' } },
-      ['Viewing read-only']),
-    div({ style: { flexGrow: 1 } }),
-    Utils.cond(
-      [!canCompute, () => h(ButtonOutline, { onClick: () => setCopying(true) }, ['copy to another workspace to edit'])],
-      () => h(ButtonOutline, {
-        href: Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
-      }, ['edit in Jupyter'])
+  Utils.useOnMount(() => {
+    const findLockedByEmail = withErrorReporting('Error loading locker information', async () => {
+      const potentialLockers = await findPotentialNotebookLockers(canShare, namespace, name, bucketName)
+      const currentLocker = potentialLockers[lockedBy]
+      setLockedByEmail(currentLocker || null)
+    })
+    findLockedByEmail()
+  })
+
+  return h(Modal, {
+    width: 530,
+    title: 'Notebook Is In Use',
+    onDismiss,
+    showButtons: false
+  }, [
+    p(lockedByEmail ? `This notebook is currently being edited by ${lockedByEmail}.` : `This notebook is currently locked because another user is editing it.`),
+    p('You can make a copy, or run it in Playground Mode to explore and execute its contents without saving any changes.'),
+    div({ style: { marginTop: '2rem' } }, [
+      h(ButtonSecondary, {
+        style: { paddingRight: '1rem', paddingLeft: '1rem' },
+        onClick: () => onDismiss()
+      }, ['CANCEL']),
```

Comment:
These take care of capitalization themselves.
```suggestion
      }, ['Cancel']),
```